src/Floor.java

```diff
@@ -1,4 +1,62 @@
-
+/**
+ * The floor class is responsible for tracking the floor number and also
+ * toggling the lights for the up/down buttons.
+ *
+ */
 public class Floor {
+	private int floorNumber;
+	private FloorButton upButton;
+	private FloorButton downButton;
+
+	/**
+	 * Constructor for Floor.
+	 * 
+	 * @param floorNumber - The current floor of the elevator.
+	 */
+	public Floor(int floorNumber) {
+		this.floorNumber = floorNumber;
+	}
+
+	/**
+	 * Turns on the Lamp depending on the direction of the elevator. If the elevator
+	 * is going up, then the lamp will turn on the up lamp.
+	 * 
+	 * @param direction
+	 * @param lampStatus
+	 */
+	public void turnButtonOn(Direction direction, boolean lampStatus) {
```

Comment:
nitpicky, but the param should be called buttonStatus. ik you're changing the lampstatus in the backend, but the end-user thinks they're turning on a button, they don't know what lampStatus is

Reply:
Yeah i agree